frontend/src/main/scala/bloop/bsp/ProjectUris.scala

```diff
@@ -0,0 +1,25 @@
+package bloop.bsp
+
+import java.net.URI
+
+import bloop.Project
+import bloop.engine.{Dag, State}
+import bloop.io.AbsolutePath
+
+object ProjectUris {
+  def getProjectDagFromUri(projectUri: String, state: State): Option[Project] = {
+    if (projectUri.isEmpty) None
+    else {
+      val uri = new URI(projectUri)
+      val query = uri.getQuery().split("&").map(_.split("="))
```

Comment:
I think that you need `getRawQuery` here, otherwise, if a value in the query contains `&` you'll split it. Same problem with `=`.